### PR TITLE
[IIO] Support device number, filled _finalize() @open sesame 4/2 15:17

### PR DESCRIPTION
**Pull request by kparichay**
1. Support device/trigger number as input, added corresponding test case as well
2. Set proper max limit on buffer capacity and frequency
3. Filled _finalize()

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 1]
```diff
@@ -519,6 +534,45 @@ gst_tensor_src_iio_get_id_by_name (const gchar * dir_name, const gchar * name,
...
+    goto error_free_filename;
+  }
+
+  if (!g_file_get_contents (filename, &file_contents, NULL, &error)) {
+    GST_ERROR ("Unable to read %s, error: %s.\n", filename, error->message);
+    g_error_free (error);
+    goto error_free_filename;
+  }
+
+  g_free (filename);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: I think these two lines are unnecessary

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 2]
```diff
@@ -1010,7 +1080,14 @@ gst_tensor_src_iio_get_property (GObject * object, guint prop_id,
...
 gst_tensor_src_iio_finalize (GObject * object)
 {
-  /** FIXME: fill this function */
+  GstTensorSrcIIO *self;
+  self = GST_TENSOR_SRC_IIO (object);
+
+  g_free (self->mode);
+  g_free (self->device.name);
+  g_free (self->trigger.name);
+
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: is it necessary to free channels list?

Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: channels list is freed in _stop()

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 1 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: done.

**Comment by kparichay** [comment]
**V2**
- Removed redundant code from gst_tensor_src_iio_get_name_by_id()

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 3]
```diff
@@ -915,10 +966,18 @@ gst_tensor_src_iio_set_property (GObject * object, guint prop_id,
       self->device.name = g_value_dup_string (value);
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: What if ```self->device.name``` is NOT null?

Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Handled.
Also added check for setting properties in null/ready state only.

**Comment by myungjoo** [comment]
Need Rebase

**Comment by kparichay** [comment]
> Need Rebase

Done

**Pull request merged**